pkg/upgrade/upgradecluster/cluster.go

```diff
@@ -47,6 +48,9 @@ type ClusterConfig struct {
 	// to expose only relevant, vetted bits of kv.DB. It'll make our tests less
 	// "integration-ey".
 	DB *kv.DB
+
+	// Cluster settings allow access to version and other settings.
+	Settings *cluster.Settings
```

Comment:
In some places, we've replaced the use of the `rpcbase.NodeDial` interface with `*nodedialer.Dialer`. So, I wonder, why can't we replace `ClusterConfig.Dialer`'s type with `nodedialer.Dialer`? Then we wouldn't need to add this `Settings` field.